modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridDhtAtomicAbstractUpdateFuture.java

```diff
@@ -387,46 +393,58 @@ final void map(ClusterNode nearNode,
         GridCacheReturn ret,
         GridNearAtomicUpdateResponse updateRes,
         GridDhtAtomicCache.UpdateReplyClosure completionCb) {
-        if (F.isEmpty(mappings)) {
-            updateRes.mapping(Collections.<UUID>emptyList());
+        try (
+            TraceSurroundings ignored =
```

Comment:
Seems like CACHE_API_DHT_UPDATE_MAP can be sometimes detached from global cache.put() tree.
Sometimes updateAllAsyncInternal0 is not called from thread that is handling near update request:
`affFut.listen(f -> updateAllAsyncInternal0(node, req, completionCb));`

Reply:
Fixed.